src/gpuarray_array_collectives.c

```diff
@@ -10,7 +10,8 @@
  */
 static inline size_t find_total_elems(const GpuArray* array) {
   unsigned int i;
-  size_t total_elems = 1;
+  size_t total_elems;
+  total_elems = 1;
```

Comment:
Yes for `for` you have to declare the iteration variable beforehand.

As for initializing on declaration, it's not a problem so we should keep doing it.